apps/web/src/app/[locale]/challenge/[slug]/submissions/[[...catchAll]]/page.tsx

```diff
@@ -40,9 +41,13 @@ export default async function SubmissionPage({ params: { id } }: Props) {
 }
 
 export type ChallengeSubmissions = NonNullable<Awaited<ReturnType<typeof getChallengeSubmissions>>>;
-const getChallengeSubmissions = cache((userId: string, challengeId: string) => {
+const getChallengeSubmissions = cache(async (userId: string, slug: string) => {
+  const challenge = await prisma.challenge.findFirstOrThrow({
```

Comment:
we used to be able to get id from the url but since its slug i need to query and get the id.

i could make slug a foreign key to submissions but unsure if thats a good idea.. id seems the only reasonable FK here